tests/test_pipeline/test_hierarchical_pipeline.py

```diff
@@ -267,3 +280,120 @@ def test_backtest_w_exog(product_level_constant_hierarchical_ts_w_exog, reconcil
     pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
     metrics, _, _ = pipeline.backtest(ts=ts, metrics=[MAE()], n_folds=2, aggregate_metrics=True)
     np.testing.assert_array_almost_equal(metrics["MAE"], 0)
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="product", source_level="market", period=1, method="PHA"),
+        BottomUpReconciliator(target_level="market", source_level="product"),
+    ),
+)
+def test_forecast_interval_presented(product_level_constant_hierarchical_ts, reconciliator):
+    ts = product_level_constant_hierarchical_ts
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=2)
+
+    pipeline.fit(ts=ts)
+    forecast = pipeline.forecast(prediction_interval=True, n_folds=1, quantiles=[0.025, 0.5, 0.975])
+    quantiles = match_target_quantiles(set(forecast.columns.get_level_values(1)))
+    assert quantiles == {"target_0.025", "target_0.5", "target_0.975"}
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="product", source_level="market", period=1, method="AHP"),
+        TopDownReconciliator(target_level="product", source_level="market", period=1, method="PHA"),
+        BottomUpReconciliator(target_level="market", source_level="product"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_forecast_prediction_intervals(product_level_constant_hierarchical_ts, reconciliator):
+    ts = product_level_constant_hierarchical_ts
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=2)
+
+    pipeline.fit(ts=ts)
+    forecast = pipeline.forecast(prediction_interval=True, n_folds=1)
+    for segment in forecast.segments:
+        target = forecast[:, segment, "target"]
+        np.testing.assert_array_almost_equal(target, forecast[:, segment, "target_0.025"])
+        np.testing.assert_array_almost_equal(target, forecast[:, segment, "target_0.975"])
+
+
+@pytest.mark.parametrize(
+    "quantiles",
+    ((0.25, 0.75), (0.1, 0.9)),
+)
+def test_forecast_get_level(product_level_constant_hierarchical_ts, quantiles):
```

Comment:
Move this test to tsdataset, here you should test it's `get_level_dataset_method`